config/routes.php

```diff
@@ -134,7 +134,8 @@ function_name: 'spar',
         function_name: 'healingShop',
         menu: Route::MENU_ACTIVITY,
         battle_ok: false,
-        village_ok: Route::VILLAGE_OR_COLOSSEUM,
+        village_ok: Route::ONLY_IN_VILLAGE,
+        village_restriction_overrides: [Route::COLOSSEUM_OVERRIDE => "true"],
```

Comment:
I think this is the point we should move away from `village_ok` entirely to a more granular `allowed_location_types` array that maps from location type => allowed or not

```suggestion
        allowed_location_types: [
            Route::LOCATION_VILLAGE,
            Route::LOCATION_COLOSSEUM
        ],
```
If this param is not provided, we could assume all location types are okay. But if provided, we only let the route trigger if the current location type matches one in this array

Reply:
I like `allowed_location-types` solution much better, building on `village_ok` logic makes this overly complicated.
I will rework this and apply to all existing routes in a follow up.